### PR TITLE
new module circexplorer2/parse, annotate

### PR DESCRIPTION
**Pull request by BarryDigby**
## PR checklist

Added two functions `parse` and `annotate` from the tool `CIRCexplorer2`. 

- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md) - I think?
- [x] If necessary, include test data in your PR.
    - circRNA test data is funky and unlikely to be of any use to others. I'm happy hosting it on the circrna branch of test-datasets.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfill software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware` -> **runs on actions**
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
      - `Conda` run will not work on actions but works perfectly fine locally.

### CODE REVIEW

File: modules/nf-core/circexplorer2/annotate/main.nf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    CIRCexplorer2 \\
+        annotate \\
+        -r $gene_annotation \\
+        -g $fasta \\
+        -b $junctions \\
+        -o ${prefix}.txt
```
Comment (jfy133) [inline modules/nf-core/circexplorer2/annotate/main.nf:30]: Missing $args

File: modules/nf-core/circexplorer2/annotate/meta.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: "circexplorer2_annotate"
+description: Annotate circRNAs detected in the output from CIRCexplorer2 parse
+keywords:
+  - circRNAs
+  - Annotate
```
Comment (jfy133) [inline modules/nf-core/circexplorer2/annotate/meta.yml:5]:
```suggestion
  - annotate
  - circexplorer
```

File: modules/nf-core/circexplorer2/annotate/meta.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+name: "circexplorer2_annotate"
+description: Annotate circRNAs detected in the output from CIRCexplorer2 parse
+keywords:
+  - circRNAs
```
Comment (jfy133) [inline modules/nf-core/circexplorer2/annotate/meta.yml:4]:
```suggestion
  - circrnas
```

File: modules/nf-core/circexplorer2/parse/main.nf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def aligner = "${fusions}".endsWith(".junction") ? "-t STAR" : "${fusions}".endsWith(".txt") ? "-t MapSplice" : "${fusions}".endsWith(".bam") ? "-t BWA" : "-t segemehl"
+    """
+    CIRCexplorer2 \\
+        parse \\
+        $aligner \\
+        $fusions \\
+        -b ${prefix}.junction.bed
```
Comment (jfy133) [inline modules/nf-core/circexplorer2/parse/main.nf:28]: Missing $args

Comment (jfy133) [inline modules/nf-core/circexplorer2/parse/main.nf:28]: Should just be `.bed`, if there is  risk of a clash, add a warning [^1]

[^1]: https://nf-co.re/docs/contributing/modules#naming-conventions

File: modules/nf-core/circexplorer2/parse/meta.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+name: "circexplorer2_parse"
+description: CIRCexplorer2 parses fusion junction files from multiple aligners to prepare them for CIRCexplorer2 annotate.
+keywords:
+  - parse
+  - circRNA
```
Comment (jfy133) [inline modules/nf-core/circexplorer2/parse/meta.yml:5]:
```suggestion
  - circrnas
  - circexplorer
```

File: modules/nf-core/circexplorer2/parse/meta.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - fusions:
+      type: file
+      description: BAM (BWA), BED (Segemehl), TXT (MapSplice), Junction (STAR).
```
Comment (jfy133) [inline modules/nf-core/circexplorer2/parse/meta.yml:23]:
```suggestion
      description: BAM (BWA), BED (Segemehl), TXT (MapSplice), or Junction (STAR) file. Aligner will be autodetected based on file suffix.
```

**Review by jfy133 — CHANGES_REQUESTED** [review]
Missing `$args` in both modules but otherwie cosmetic changes!

**Review by ramprasadn — APPROVED** [review]
LGTM! 🚀

**Review by nvnieuwk — APPROVED** [review]

**Review by jfy133 — APPROVED** [review]
A few more keywords would be useful for findability but otherwise good to og!

**Pull request merged**